content-services/latest/tutorial/sso/ldap.md

```diff
@@ -0,0 +1,296 @@
+Alfresco Process Services (APS) has two sets of properties that need to be configured to setup SSO. One set synchronizes APS with an LDAP directory and the other set configures the connection with Keycloak.
+
+1. Configuration for LDAP synchronization can be achieved manually for WAR file deployments using the `activiti-ldap-properties` file or reference an external file for Docker and Kubernetes deployments:
+
+    | Property | Description |
+    | -------- | ----------- |
+    | ldap.authentication.enabled | Sets whether LDAP authentication is enabled. This needs to be `false` as LDAP is only being used for user synchronization, for example `false`. |
+    | ldap.authentication.java.naming.provider.url | The URL of the LDAP instance, for example `ldaps://ldap.example.com:636`. |
+    | ldap.synchronization.java.naming.security.principal | The user used to access the LDAP directory to perform the synchronization, for example `uid=admin,ou=system`. |
+    | ldap.synchronization.java.naming.security.credentials | The password for the user set in `ldap.synchronization.java.naming.security.principal`, for example `secret`. |
+    | ldap.synchronization.full.enabled | Sets whether full LDAP synchronization is enabled or not, for example `true`. |
+    | ldap.synchronization.full.cronExpression | The cron expression describing how often the full synchronization should run, for example `0 0 0 * * ?`. |
+    | ldap.synchronization.differential.enabled | Sets whether differential LDAP synchronization is enabled or not, for example `true`. |
+    | ldap.synchronization.differential.cronExpression | The cron expression describing how often the differential synchronization should run, for example `0 0 */4 * * ?`. |
+    | ldap.synchronization.userSearchBase | The section of the LDAP directory to restrict user synchronization to, for example `ou=users,dc=alfresco,dc=com`. |
+    | ldap.synchronization.groupSearchBase | The section of the LDAP directory to restrict group synchronization to, for example `ou=groups,dc=alfresco,dc=com`. |
+
+2. Configuration with the Alfresco Process Services can be achieved manually for WAR file deployments using the `activiti-identity-service.properties` or reference an external file for Docker and Kubernetes deployments:
+
+    | Property | Description |
+    | -------- | ----------- |
+    | keycloak.enabled | Sets whether Process Services will use Keycloak to authenticate against, for example `true`. |
+    | keycloak.realm | The realm name configured in Keycloak for the Alfresco applications, for example `alfresco`. |
+    | keycloak.auth-server-url | Keycloak's base URL, for example `https://keycloak.example.com/auth`. |
+    | keycloak.ssl-required | Sets whether SSL is mandatory for access or not, for example `all`. |
+    | keycloak.resource | The **Client ID** set up in Keycloak for Process Services. The client needs to exist underneath the realm set for `keycloak.realm` or `KEYCLOAK_REALM`, for example `alfresco`. |
```

Comment:
The original documentation stated:

`The client needs to exist underneath the realm set for keycloak.realm or IDENTITY_SERVICE_REALM, for example alfresco`

@balsarori @jamalkm was this a mistake in the first place? Was this supposed to be _"`keycloak.realm` or `KEYCLOAK_REALM`"_ in the first place?

I assume the new documentation wording makes sense if we're referring to [binding from environment variables](https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config.typesafe-configuration-properties.relaxed-binding.environment-variables) via Spring, I assume it wouldn't if there is a custom mechanism to specifically look for an `IDENTITY_SERVICE_REALM` env var.

@mbujak-hyland please wait for @balsarori or @jamalkm confirmation related to this question.

Reply:
there used to be a script in APS that  checks if the environment variable IDENTITY_SERVICE_REALM is set to a non-empty value. If it is, it replaces the keycloak.realm value with that one.

@dsibilio there used to be a script in APS that checks whether the environment variable `IDENTITY_SERVICE_REALM` is set to a non-empty value. If it is, it replaces the `keycloak.realm` value with that one. This is for @balsarori to confirm.

Reply:
@moberoi2

Reply:
There is no `KEYCLOAK_REALM` env var. In APS 2, there was a property called `keycloak.realm` which is set by setting the env var `IDENTITY_SERVICE_REALM`. In APS 24.x the property name was changed to `activiti.identity-service.realm` the env var remains `IDENTITY_SERVICE_REALM`

Reply:
Quoting @balsarori:

> there is no `KEYCLOAK_REALM` env var. In APS 2, there was a property `keycloak.realm` which is set by setting the `IDENTITY_SERVICE_REALM`. In APS the property name was changed to `activiti.identity-service.realm`

@mbujak-hyland I also confirmed that both for APS 2 and APS 24 the env var is always `IDENTITY_SERVICE_REALM` and can't be renamed in the docs to `KEYCLOAK_REALM`